rust/candid/src/parser/typing.rs

```diff
@@ -261,6 +264,88 @@ fn check_actor(env: &Env, actor: &Option<IDLType>) -> Result<Option<Type>> {
     }
 }
 
+pub(crate) fn interpret_number(number: IDLValue, typ: IDLType) -> Result<IDLValue> {
+    let number = match number {
+        IDLValue::Number(number) => number,
+        _ => Err(Error::msg("Non-number in type annotated value"))?,
+    };
+
+    fn error<E: ToString>(err: E) -> Error {
+        Error::msg(err.to_string())
+    }
+
+    match typ {
+        IDLType::PrimT(prim_typ) => {
+            match prim_typ {
+                PrimType::Nat => {
+                    let biguint = BigUint::from_str(number.as_str()).map_err(error)?;
+                    Ok(IDLValue::Nat(Nat(biguint)))
+                }
+                PrimType::Int => {
+                    let bigint = BigInt::from_str(number.as_str()).map_err(error)?;
+                    Ok(IDLValue::Int(Int(bigint)))
+                }
+                // NOTE: Concrete syntax for numbers is not specified in candid spec, but the
```

Comment:
Spec says it’s same as Motoko, which I believe is the same as Wasm’s `.wat`, I believe, i.e. https://webassembly.github.io/spec/core/text/values.html.

Reply:
Where do we say that Motoko's nat/int syntax is the same as .wat syntax? I checked https://sdk.dfinity.org/docs/language-guide/motoko-grammar.html but couldn't see it.

Reply:
Might have been “personal communication” (or just me misremembering)